### ISSUE TITLE
Failed to rename backups, aborting upload: unexpected EOF

### ISSUE THREAD

**Issue by jamesmyatt** [jcwillox/hassio-rclone-backup#34]
### The problem

I'm getting the error message "unexpected EOF" which is causing the upload to fail. The main issue is that this error message isn't very useful.

Looking at the source code, I suspect it's coming from [here](https://github.com/jcwillox/hassio-rclone-backup/blob/9d27a6c3d88643f491799dc78cc2cb79051a28c2/rclone_backup/scheduler/backups.go#LL86C22-L86C22), which I suspect means that one of the backup files is corrupt, probably because of running out of disk space when making the backup. If that's right, then the error message really ought to say which file it was reading when it got the "unexpected EOF", which would allow me to find and remove the corrupt backup.

Second, if this is correct, then it would be great if the upload could also continue without the corrupt file.

### What version of Rclone Backup has the issue?

3.0.1

### What version of Home Assistant are you running?

2023.5.4

### What type of installation are you running?

Home Assistant OS

### If you're running HA OS/Supervised, what version of the Supervisor are you running?

2023.06.1

### Anything in the logs that might be useful for us?

```python3
[2023-06-09] [04:45:00] [INFO]: running "Sync Onedrive"; /backup -> onedrive:Backup/home-assistant/main
[2023-06-09] [04:45:00] [ERROR]: failed to rename backups, aborting upload: unexpected EOF
```

**Comment by FillR2** [comment]
SAme error here

**Comment by jcwillox** [comment]
It probably means you have some corrupted tar files in your backups folder, or non-backup tar files in there, maybe try temporarily removing all the backups except known good ones, or as @FillR2 said add `no_rename: true` to disable the renaming functionality.

**Comment by angristan** [comment]
I agree that a more resilient approach would be nice here; right now, it fails to sync any backup until the corrupted tar is removed 🤔 (and renaming is nicer!)
